forage-android/src/main/java/com/joinforage/forage/android/LDManager.kt

```diff
@@ -72,17 +85,54 @@ internal object LDManager {
         }
         val contextKind = ContextKind.of(LDContextKind.SERVICE)
         val context = LDContext.create(contextKind, LDContexts.ANDROID_CONTEXT)
-        val client = LDClient.init(app, ldConfig, context, 0)
+        client = LDClient.init(app, ldConfig, context, 0)
+
+        internalLogger = logger
+    }
+
+    // vaultType is instantiated lazily and is a singleton. Once we set the vault type once, we don't
+    // want to overwrite it! We must take in the application as a parameter, which means that a
+    // ForagePINEditText must be rendered before any of the ForageSDKApi functions are called.
+    internal fun getVaultProvider(): VaultType {
+        if (vaultType != null) {
+            return vaultType as VaultType
+        }
         // default to 100% VGS usage in case LD flag retrieval fails
-        val vaultPercent = client.doubleVariation(LDFlags.VAULT_PRIMARY_TRAFFIC_PERCENTAGE_FLAG, 0.0)
+        val vaultPercent =
+            client?.doubleVariation(LDFlags.VAULT_PRIMARY_TRAFFIC_PERCENTAGE_FLAG, 0.0)
         internalLogger.i("[LaunchDarkly] Vault percent of $vaultPercent return from LD")
         val randomNum = Math.random() * 100
-        vaultType = if (randomNum < vaultPercent) {
+
+        vaultType = if (randomNum < (vaultPercent ?: 0.0)) {
             VaultType.BT_VAULT_TYPE
         } else {
             VaultType.VGS_VAULT_TYPE
         }
         internalLogger.i("[LaunchDarkly] Vault type set to $vaultType")
         return vaultType as VaultType
     }
+
+    internal fun getPollingIntervalsFeatureFlag(): IntArray {
+        if (pollingIntervals != null) {
+            return pollingIntervals as IntArray
+        }
+
+        var pollingList = IntArray(0)
+
+        val jsonIntervals =
+            client?.jsonValueVariation(LDFlags.ISO_POLLING_WAIT_INTERVALS, LDValue.buildObject().put("intervals", LDValue.buildArray().add(1000).build()).build())
```

Comment:
What will be the behavior of the default value if it's the array `[1000]`? Are we only going to poll once after 1 second?

Reply:
I just made the default LDValue a list of size 1 because it seems so verbose to have to do `.add(1000)` 10 times. Lemme see if there's an easier way to define a list rather than doing something like `.add(1000).add(1000).add(1000).add(1000).add(1000).add(1000).add(1000)`

Reply:
I was able to get it to work by doing 
```
        val defaultVal = LDValue.buildObject().put("intervals",
            LDValue.Convert.Integer.arrayFrom(listOf(1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000))).build()
```
This way it is more explicit what the default value is